### PR TITLE
[firebase_crashlytics] Migrate FlutterErrorDetails

### PR DESCRIPTION
**Pull request by mklim**
## Description

Update our handling of the struct to work on both flutter stable and
master. Tested on both master and stable. The plugin itself works on
both, but our unit tests now only run on master.

## Related Issues

flutter/flutter#31962

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by mklim** [comment]
This should also fix CI.

File: packages/firebase_crashlytics/test/firebase_crashlytics_test.dart [hunk 1]
```diff
@@ -32,10 +33,7 @@ void main() {
         exception: 'foo exception',
         stack: StackTrace.current,
         library: 'foo library',
-        context: 'foo context',
-        informationCollector: (StringBuffer information) {
-          information.writeln('foo information');
-        },
+        context: DiagnosticsNode.message('foo context'),
```
Comment (jacob314) [inline packages/firebase_crashlytics/test/firebase_crashlytics_test.dart]: nit: use ErrorDescription('foo context')
as that is the idiomatic version to use for error collection.

File: packages/firebase_crashlytics/test/firebase_crashlytics_test.dart [hunk 2]
```diff
@@ -32,10 +33,7 @@ void main() {
         exception: 'foo exception',
         stack: StackTrace.current,
         library: 'foo library',
-        context: 'foo context',
-        informationCollector: (StringBuffer information) {
```
Comment (jacob314) [inline packages/firebase_crashlytics/test/firebase_crashlytics_test.dart]: strange the test still passes with these lines removed.

File: packages/firebase_crashlytics/lib/src/firebase_crashlytics.dart [hunk 3]
```diff
@@ -47,7 +47,10 @@ class Crashlytics {
           // ignore: strong_mode_implicit_dynamic_method
           await channel.invokeMethod('Crashlytics#onError', <String, dynamic>{
         'exception': details.exceptionAsString(),
-        'context': details.context,
+        // FlutterErrorDetails.context has been migrated from a String to a
+        // DiagnosticsNode. Coerce it to a String here in a way that will work
+        // on both Strings and the new DiagnosticsNode values. See https://groups.google.com/forum/#!topic/flutter-announce/hp1RNIgej38
+        'context': "${details.context}",
```
Comment (kroikie) [inline packages/firebase_crashlytics/lib/src/firebase_crashlytics.dart]: Nit: should this be single quotes?

Comment (collinjackson) [inline packages/firebase_crashlytics/lib/src/firebase_crashlytics.dart]: nit: I would normally use single quotes

**Pull request merged**